pkg/options/patchtmpl/patch.go

```diff
@@ -41,15 +41,23 @@ type applier struct {
 	// If includeTemplates is true, applied patch templates will be included in the
 	// component objects.
 	includeTemplates bool
+
+	// Set the template options
+	templateOpts []string
 }
 
 // NewApplier creates a new options applier instance. The filter indicates
 // keep-only options for what subsets of patches to look for.
-func NewApplier(pt *PatcherScheme, opts *filter.Options, includeTemplates bool) options.Applier {
+func NewApplier(pt *PatcherScheme, opts *filter.Options, includeTemplates bool, templateOpts ...string) options.Applier {
+	if templateOpts == nil || len(templateOpts) == 0 {
```

Comment:
Why is a length 0 templateOpts invalid?

Reply:
We want it to error by default, which is the previous behavior. So if the caller doesn't do anything, this will behave just like it did before.